### PR TITLE
Niganesh/protocol changes in request

### PR DESCRIPTION
**Pull request by nithyaganeshng**
**Problem:**
First set of changes for SIOP v2 and to be compliant with PE specs which includes replacing scope and response_mode. It also includes supporting id in presenation_definition and additional parameters in registration parameters.

**Solution:**
Used library A to complete scenario X with quality and style.

**Validation:**
Please include here how it was verified that the PR works as intended.

**Type of change:**
- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/Response.kt [hunk 1]
```diff
@@ -30,6 +30,7 @@ class IssuanceResponse(override val request: IssuanceRequest) :
 class PresentationResponse(override val request: PresentationRequest) :
     Response(request, request.content.clientId ?: throw ValidatorException("No audience in presentation request")) {
     val requestedVcPresentationSubmissionMap: RequestedVcPresentationSubmissionMap = mutableMapOf()
+    val requestedVcPresentationDefinitionId: String = request.getPresentationDefinition().id
```
Comment (dangodb) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/Response.kt:33]: Are we making changes to the issuance protocol?

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationResponseClaims.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ import kotlinx.serialization.Serializable
 @Serializable
 data class PresentationResponseClaims(
     @SerialName("presentation_submission")
-    val presentationSubmission: PresentationSubmission = PresentationSubmission(emptyList()),
+    val presentationSubmission: PresentationSubmission = PresentationSubmission("","",emptyList()),
```
Comment (dangodb) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationResponseClaims.kt:20]: String.Empty?

File: sdk/src/main/java/com/microsoft/did/sdk/util/Constants.kt [hunk 3]
```diff
@@ -13,8 +13,11 @@ object Constants {
     const val SELF_ISSUED = "https://self-issued.me"
     const val PURE_ISSUANCE_FLOW_VALUE = "create"
     const val RESPONSE_TYPE = "id_token"
-    const val RESPONSE_MODE = "form_post"
-    const val SCOPE = "openid did_authn"
+    const val RESPONSE_MODE = "post"
+    const val SCOPE = "openid"
+    const val SUBJECT_IDENTIFIER_TYPE_DID = "did"
+    const val DID_METHODS_SUPPORTED = "did:ion:"
+    const val ALGORITHM_SUPPORTED_IN_VP = "ES256"
```
Comment (dangodb) [inline sdk/src/main/java/com/microsoft/did/sdk/util/Constants.kt:20]: Shouldn't this be ES256K?

**Review by dangodb — APPROVED** [review]

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/Registration.kt [hunk 4]
```diff
@@ -28,5 +29,15 @@ data class Registration(
...
+
+    @SerialName("subject_identifier_types_supported")
+    val subjectIdentifierTypesSupported: List<String> = emptyList(),
+
+    @SerialName("did_methods_supported")
+    val didMethodsSupported: List<String> = emptyList(),
+
+    @Serializable(with = PresentationRequestFormatSerializer::class)
+    @SerialName("vp_formats")
+    val format: List<String> = emptyList()
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/Registration.kt:42]: nit, why not `vpFormats`?

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/presentationexchange/PresentationRequestFormatSerializer.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+
+/**
+ * This class serializes/deserializes format property in Presentation Request.
+ * The format is a Key/Value pair where key can be "jwt_vp" and value is a list of String.
+ * @see [Presentation Exchange] (https://identity.foundation/presentation-exchange/#presentation-submission)
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/presentationexchange/PresentationRequestFormatSerializer.kt:23]: I don't quite understand how this serializer is matching what is found at this link. Is the link up to date? I'm probably not looking at the right place. You can explain it to me in person maybe :)

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationResponseClaims.kt (hunk 2 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/credential/service/models/oidc/PresentationResponseClaims.kt]: removed it.

**Review by symorton — APPROVED** [review]

**Pull request merged**